source/puppet/4.10/functions_ruby_implementation.md

```diff
@@ -77,28 +77,28 @@ end
 
 ## Calling other functions
 
-If you want to call another Puppet function (like `include`) from inside a function, use the special `call_function(name, args, &block)` method.
+If you want to call another Puppet function (like `include`) from inside a function, use the special `call_function(name, *args, &block)` method.
 
 ``` ruby
 # Flatten an array of arrays of strings, then pass it to include:
 def include_nested(array_of_arrays)
-  call_function('include', array_of_arrays.flatten)
+  call_function('include', *array_of_arrays.flatten)
 end
 ```
 
 * The first argument must be the name of the function to call, as a string.
-* The second argument must be an array containing any arguments to the function.
-* The third argument can be a Ruby Proc, or a Puppet [lambda][lambdas] previously captured as a Proc (see above). You can also provide a block of Ruby code using the normal block syntax.
+* The next arguments can be any data type that the called function accepts. They will be passed through directly as arguments to the called function.
```

Comment:
Not sure why the phrase "passed through directly" is saying - it may not be true as there may be type casting involved if a parameter type is of `Init` type. Suggest taking out those words.
(Also, it opens up the question "what is the difference between direct and indirect passing of arguments - is there even such a thing?").